src/main/java/org/zalando/nakadi/controller/EventPublishingController.java

```diff
@@ -100,19 +106,23 @@ private ResponseEntity postEventInternal(final String eventTypeName,
         }
     }
 
+    private void reportSLOs(final long startingNanos, final int totalSizeBytes, final int eventCount) {
+        final long msSpent = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startingNanos);
```

Comment:
Probably, you transform from nanos to millis you lose precision. Maybe just take System.currentTimeMillis() ?